charts/DownloadTab.tsx

```diff
@@ -146,6 +159,95 @@ export class DownloadTab extends React.Component<DownloadTabProps> {
         }
     }
 
+    // Here's where the actual CSV is made
+    @computed get csvBlob() {
```

Comment:
I feel like we should rather move this to a different class instead of duplicating it - this is complex code that doesn't belong in a view class at all, and having it in two places makes it worse because the implementations can diverge in the future if we're not careful.

I assume the duplication is meant to be only temporary, but it still makes sense to move this logic out of the view itself (I don't have an idea for where to move it, though)

Reply:
Yes, the duplication is temporary, and the code is a direct transfer from DataTab (another view class). I agree it can be moved into its own class e.g. "CSVGenerator", and will do that.